FS-UAE initial recipe

moochris opened the pull request:
Thanks to miqlas for creating the patchset to initally get things working. JIT is also working now.

moochris commented:
Argh, something went horribly wrong and some commits from another branch got included...

pulkomandy commented:
The description seems a bit long. Maybe some parts could be removed? (at least the bit about the license since that's already in the package metadata, but I'd remove a lot more).

moochris commented:
Yes, it's way too long - it's the description from the homepage. I'm not sure which bits to edit out or keep!

Chris

On 7 August 2017 20:42:43 BST, Adrien Destugues <notifications@github.com> wrote:
>The description seems a bit long. Maybe some parts could be removed?
>(at least the bit about the license since that's already in the package
>metadata, but I'd remove a lot more).
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/haikuports/haikuports/pull/1566#issuecomment-320760583

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

diversys commented:
You don't need to bump REVISION if you simply enable architecture since the recipe itself doesn't change.

diversys commented:
Ah, I spoke too soon. I now see that you edited a patch.

moochris commented:
Sorry - I removed the unwanted changes by merging upstream/master to my haikuports/master and then merging into my branch again. Should just be FS-UAE stuff now.

I shortened the description and removed the trailling space.

pulkomandy commented:
Still nitpicking on the too long description text:

- Configuration GUI: is that present in the package? I don't see it in the provides. If not, I would remove it
- Operating systems support: I'd say we are only concerned with the Haiku version, so we could remove that

Other than that, the changes look mostly ok. No secondary arch support, but we can add that later.

korli commented on app-emulation/fs-uae/patches/fs_uae-2.8.3.patchset:
```diff
@@ -0,0 +1,503 @@
+         return 1;
+     }
++    
++    char* data_path = "/system/data/fs-uae";
```
This shouldn't be hardcoded. here is a good example: https://github.com/haikuports/haikuports/blob/master/games-action/teeworlds/patches/teeworlds_0.6.3.patchset#L337

moochris replied:
Thanks - I'm using the right method now.

korli commented on app-emulation/fs-uae/fs_uae-2.8.3.recipe:
```diff
@@ -0,0 +1,119 @@
+SOURCE_DIR="fs-uae-$portVersion"
+PATCHES="fs_uae-$portVersion.patchset"
+
+ARCHITECTURES="?x86 !x86_gcc2 x86_64"
```
You probably want to add: SECONDARY_ARCHITECTURES="?x86"

moochris replied:
Done

moochris commented:
@pulkomandy @korli - ok, will fix up this evening

_Sent from my Lge LG-K500n using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

miqlas commented on app-emulation/fs-uae/patches/fs_uae-2.8.3.patchset:
```diff
@@ -0,0 +1,503 @@
++#include <posix/endian.h>
++#include <support/ByteOrder.h>
++
++#define __bswap16(x)	__swap_int16(x)
```
Unneded with BSD_SOURCE

miqlas commented on app-emulation/fs-uae/patches/fs_uae-2.8.3.patchset:
```diff
@@ -0,0 +1,503 @@
++
++}
++
++#if 0
```
You should clean this file up, it is full with unneded/commented stuff.

moochris commented:
@miqlas i don't have the spare time, I'm afraid. I'll see what I can get done in the next few weeks.

_Sent from my Lge LG-K500n using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

moochris commented:
@pulkomandy OK, hopefully the description is OK now. I added the secondary arch bits.

moochris commented on app-emulation/fs-uae/fs_uae-2.8.3.recipe:
```diff
@@ -0,0 +1,119 @@
+	fs_uae = $portVersion
+	cmd:fs_uae
+	cmd:fs_uae_device_helper
+	add_on:fs_uae.dat
```
Absolutely, yes :) It's needed for all the images and other resources. I added support to load it from the right location. It tries to load it from the current working directory, so you wouldn't have noticed if you ran fs-uae in the build area.

moochris commented on app-emulation/fs-uae/fs_uae-2.8.3.recipe:
```diff
@@ -0,0 +1,119 @@
+
+REQUIRES="
+        haiku
+        mesa_swpipe
```
Not in x86_64 it wasn't - I had to install

moochris commented on app-emulation/fs-uae/fs_uae-2.8.3.recipe:
```diff
@@ -0,0 +1,119 @@
+REQUIRES="
+        haiku
+        mesa_swpipe
+        lib:libx11
```
It's still needed for some things, as far as I know

moochris commented on app-emulation/fs-uae/fs_uae-2.8.3.recipe:
```diff
@@ -0,0 +1,119 @@
+        lib:libmpeg2
+        lib:libpng16
+        lib:libsdl2
+        lib:libz
```
This is now ordered.

korli commented:
@moochris I've cleaned up what I could, the patchset is still to be reworked (unneeded patch with _BSD_SOURCE, etc)

moochris commented on app-emulation/fs-uae/patches/fs_uae-2.8.3.patchset:
```diff
@@ -0,0 +1,503 @@
+ 
+ #include <stdint.h>
+ 
++#ifdef __HAIKU__
```
OK, this can now be done thanks to @jessicah with changes so that -D_BSD_SOURCE allows including <endian.h> with the endian conversion macros.
I will update the patch shortly.

moochris commented:
All issues now addressed, please review.

miqlas commented:
I think it is almost okay. Can we have an Amiga beachball as icon?

moochris commented:
Hah :)
No, the icon should be on the launcher once I've added the recipe - nearly done:
https://discuss.haiku-os.org/uploads/default/original/2X/2/2ed73b99ba8a36e9d8344d92d2177cc550076691.jpg

moochris commented:
@miqlas please can you mark your review as changes accepted or something? It still reads 'changes requested' for the PR. :)

miqlas commented:
@moochris have you checked if support/ByteOrder.h required?

moochris commented:
@miqlas good spot - the other include was also now redundant. Now fixed.

moochris commented:
Is there anything else I need to do for this? I think all the reviews have been responded to.
